### PR TITLE
[python] add multi-select dialog

### PR DESCRIPTION
**Pull request by tamland**
Exposes the multi-select dialog to python. Uses indices for consistency with the single select api.

### CODE REVIEW

**Comment by hudokkow** [comment]
jenkins build this please

**Comment by tamland** [comment]
@mkortstiege /@xhaggi for review please. There are core changes as well here.

**Comment by mkortstiege** [comment]
Really not sure about the codegenerator one but changes looking good to me otherwise. Please don't forget to bump the API version after this went in or  the merge window closes.

**Comment by Montellese** [comment]
I'll try to take a look at the codegenerator and Python API changes over the weekend.

**Comment by tamland** [comment]
After checking the API I believe `Player::getAvailableSubtitleStreams` and `Player::getAvailableAudioStreams` are the only two returning vector pointer. They would change from returning empty list to None on failure with the generator change.

Either we consider that a bug and change the API, or change the c++ code to return empty vector instead of null (with the generator change we can now do both).

**Comment by xhaggi** [comment]
apart from the minors this looks good, thanks :+1:

**Comment by Montellese** [comment]
IMO in case of `Player::getAvailableSubtitleStreams` and `Player::getAvailableAudioStreams` I don't see what the advantage of returning `None` over an empty array/list would be. It doesn't have a different meaning and IMO getting an empty list is more intuitive. I don't even understand why it's implemented to return a pointer-to-vector instead of just vector.

**Comment by tamland** [comment]
Updated. all good?

**Comment by xhaggi** [comment]
looks good from my side

File: xbmc/interfaces/legacy/Player.cpp [hunk 1]
```diff
@@ -426,26 +426,26 @@ namespace XBMCAddon
       }
     }
 
-    std::vector<String>* Player::getAvailableSubtitleStreams()
+    std::vector<String> Player::getAvailableSubtitleStreams()
```
Comment (jimfcarroll) [inline xbmc/interfaces/legacy/Player.cpp]: Was this a leak before?

Comment (tamland) [inline xbmc/interfaces/legacy/Player.cpp]: @jimfcarroll What do you mean? They were freed were they not? Otherwise we have lots of leaks..

Comment (jimfcarroll) [inline xbmc/interfaces/legacy/Player.cpp]: Yeah. I'm just wondering why the return type was changed from a pointer (and if that even works). I guess I was assuming someone thought it was a leak - which I don't think it is but I was trying to figure out why someone might have

Comment (tamland) [inline xbmc/interfaces/legacy/Player.cpp]: Just changed it to guarantee it doesn't return null pointer. Does it make any difference if it returns value or pointer?

Comment (jimfcarroll) [inline xbmc/interfaces/legacy/Player.cpp]: Not if it works. :-) It's less efficient.

**Comment by tamland** [comment]
jenkins build this please

**Comment by tamland** [comment]
jenkins build this please and please make an effort this time

File: xbmc/interfaces/legacy/Player.cpp (hunk 1 above)
Comment (tamland) [inline xbmc/interfaces/legacy/Player.cpp]: @jimfcarroll Are you certain it won't leak? Coverity seems to think it does.

**Comment by popcornmix** [comment]
There is a bug reported on nightly build:
"There is a bug in latest releases that user can not change groups under TV/group (slides from right) the groups (view options). Even if you select it, it does not change."

I'm just guessing but this looked like the most likely commit that day to cause it. 
http://forum.kodi.tv/showthread.php?tid=231092&pid=2084729#pid2084729
http://forum.kodi.tv/showthread.php?tid=231092&pid=2085191#pid2085191

**Comment by roee88** [comment]
An option to specify pre-selected items would make this more useful.

**Pull request merged**